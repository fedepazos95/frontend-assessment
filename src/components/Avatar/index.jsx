import React from 'react';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';
import styles from './avatarStyle';

const useStyles = makeStyles(styles);

export default function Avatar({ image, alt }) {
  const classes = useStyles();

  return (
    <img
      src={image}
      alt={alt}
      className={`${classes.rounded} ${classes.fluid}`}
    />
  );
}

Avatar.defaultProps = {
  alt: '',
};

Avatar.propTypes = {
  image: PropTypes.string.isRequired,
  alt: PropTypes.string,
};
