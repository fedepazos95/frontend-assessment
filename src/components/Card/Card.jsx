import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { makeStyles } from '@material-ui/core/styles';

import styles from './cardStyle';

const useStyles = makeStyles(styles);

export default function Card({ className, children }) {
  const classes = useStyles();
  const cardClasses = classNames({
    [classes.card]: true,
    [className]: className !== undefined,
  });
  return (
    <div className={cardClasses}>
      {children}
    </div>
  );
}

Card.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};
