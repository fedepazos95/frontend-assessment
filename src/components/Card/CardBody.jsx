import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

// Material-UI Core Components
import { makeStyles } from '@material-ui/core/styles';
// Style
import styles from './cardBodyStyle';

const useStyles = makeStyles(styles);

export default function CardBody({ className, children }) {
  const classes = useStyles();
  const cardBodyClasses = classNames({
    [classes.cardBody]: true,
    [className]: className !== undefined,
  });

  return (
    <div className={cardBodyClasses}>
      {children}
    </div>
  );
}

CardBody.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
};
