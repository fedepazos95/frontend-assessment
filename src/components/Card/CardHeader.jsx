import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

// Material-UI Core Components
import { makeStyles } from '@material-ui/core/styles';
// Style
import styles from './cardHeaderStyle';

const useStyles = makeStyles(styles);

export default function CardHeader({
  className, children, color,
}) {
  const classes = useStyles();
  const cardHeaderClasses = classNames({
    [classes.cardHeader]: true,
    [classes[`${color}CardHeader`]]: color,
    [className]: className !== undefined,
  });

  return (
    <div className={cardHeaderClasses}>
      {children}
    </div>
  );
}

CardHeader.propTypes = {
  className: PropTypes.string,
  color: PropTypes.oneOf([
    'primary',
  ]),
  children: PropTypes.node,
};
