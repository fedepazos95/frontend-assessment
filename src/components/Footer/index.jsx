import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

// Material-UI Core Components
import { makeStyles } from '@material-ui/core/styles';
// Style
import styles from './footerStyle';

const useStyles = makeStyles(styles);

export default function Footer({ whiteFont }) {
  const classes = useStyles();
  const footerClasses = classNames({
    [classes.footer]: true,
    [classes.footerWhiteFont]: whiteFont,
  });

  return (
    <footer className={footerClasses}>
      <div className={classes.container}>
        <div className={classes.right}>
          &copy;
          {`${1900 + new Date().getYear()} Frontend Assessment - by Federico Pazos`}
        </div>
      </div>
    </footer>
  );
}

Footer.propTypes = {
  whiteFont: PropTypes.bool,
};
