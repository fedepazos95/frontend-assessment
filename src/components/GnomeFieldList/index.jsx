/* eslint-disable react/no-array-index-key */
import React from 'react';
import PropTypes from 'prop-types';
import {
  List, ListItem, ListItemAvatar, Avatar, ListItemText,
} from '@material-ui/core';

export default function GnomeJobs({ title, list, icon }) {
  return (
    <List>
      <h3>{title}</h3>
      {
        list.map((job, key) => (
          <ListItem key={key}>
            <ListItemAvatar>
              <Avatar>
                {icon}
              </Avatar>
            </ListItemAvatar>
            <ListItemText
              primary={job}
            />
          </ListItem>
        ))
      }
    </List>
  );
}

GnomeJobs.propTypes = {
  title: PropTypes.string,
  list: PropTypes.arrayOf(PropTypes.string),
  icon: PropTypes.node,
};
