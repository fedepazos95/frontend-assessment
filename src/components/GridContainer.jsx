import React from 'react';
import PropTypes from 'prop-types';

import { makeStyles } from '@material-ui/core/styles';
import { Grid } from '@material-ui/core';

const styles = {
  grid: {
    marginRight: '-15px',
    marginLeft: '-15px',
    width: '100%',
  },
};

const useStyles = makeStyles(styles);

export default function GridContainer(props) {
  const classes = useStyles();
  const { children, className } = props;
  return (
    <Grid container className={`${classes.grid} ${className}`}>
      {children}
    </Grid>
  );
}

GridContainer.defaultProps = {
  className: '',
};

GridContainer.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
};
