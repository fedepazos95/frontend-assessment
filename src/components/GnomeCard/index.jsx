import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';

// Material-UI Core Components
import { makeStyles } from '@material-ui/core/styles';
// Components
import Card from 'components/Card/Card';
import CardBody from 'components/Card/CardBody';
import CardHeader from 'components/Card/CardHeader';
// Style
import styles from './gnomeCardStyle';

const useStyles = makeStyles(styles);

export default function GnomeCard({
  headerColor, data, title,
}) {
  const classes = useStyles();
  const cardTitle = classNames({
    [classes.cardTitle]: true,
  });
  return (
    <Card>
      <CardHeader color={headerColor}>
        {title !== undefined ? <div className={cardTitle}>{title}</div> : null}
      </CardHeader>
      <CardBody>
        {data.content}
      </CardBody>
    </Card>
  );
}

GnomeCard.propTypes = {
  headerColor: PropTypes.oneOf([
    'primary',
  ]),
  data: PropTypes.shape({
    content: PropTypes.object.isRequired,
  }),
  title: PropTypes.string,
};
