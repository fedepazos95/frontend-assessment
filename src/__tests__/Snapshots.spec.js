/* eslint-disable react/jsx-filename-extension */
import React from 'react';
import renderer from 'react-test-renderer';

// Components
import Avatar from 'components/Avatar';
import Card from 'components/Card/Card';
import CardHeader from 'components/Card/CardHeader';
import CardBody from 'components/Card/CardBody';
import Footer from 'components/Footer';
import GnomeCard from 'components/GnomeCard';
import GnomeFieldList from 'components/GnomeFieldList';
import GridContainer from 'components/GridContainer';
import GridItem from 'components/GridItem';
import Header from 'components/Header';
import Parallax from 'components/Parallax';
import Stats from 'components/Stats';

describe('Avatar', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Avatar />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Card', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Card />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('CardHeader', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<CardHeader />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('CardBody', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<CardBody />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Footer', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Footer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('GnomeCard', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<GnomeCard data={
      {
        content: (
          <div />
        ),
      }
    }
    />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('GnomeFieldList', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<GnomeFieldList list={[]} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('GridContainer', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<GridContainer />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('GridItem', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<GridItem />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Header', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Header />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Parallax', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Parallax />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Stats', () => {
  test('snapshot renders', () => {
    const component = renderer.create(<Stats />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
