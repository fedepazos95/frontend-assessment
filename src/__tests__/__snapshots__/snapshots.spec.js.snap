// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar snapshot renders 1`] = `
<div
  className="makeStyles-rounded-2 makeStyles-fluid-1"
  style={
    Object {
      "backgroundImage": "url(undefined)",
      "backgroundSize": "cover",
      "minHeight": undefined,
    }
  }
/>
`;

exports[`Card snapshot renders 1`] = `
<div
  className="makeStyles-card-3"
/>
`;

exports[`CardBody snapshot renders 1`] = `
<div
  className="makeStyles-cardBody-6"
/>
`;

exports[`CardHeader snapshot renders 1`] = `
<div
  className="makeStyles-cardHeader-4"
/>
`;

exports[`Footer snapshot renders 1`] = `
<footer
  className="makeStyles-footer-8"
>
  <div
    className="makeStyles-container-10"
  >
    <div
      className="makeStyles-right-7"
    >
      Â©
      2019 Frontend Assessment - by Federico Pazos
    </div>
  </div>
</footer>
`;

exports[`GnomeCard snapshot renders 1`] = `
<div
  className="makeStyles-card-3 makeStyles-card-11"
>
  <div
    className="makeStyles-cardHeader-4"
  />
  <div
    className="makeStyles-cardBody-6"
  >
    <div />
  </div>
</div>
`;

exports[`GnomeFieldList snapshot renders 1`] = `
<ul
  className="MuiList-root MuiList-padding"
>
  <h3 />
  <div>
    This gnome has no undefined :(
  </div>
</ul>
`;

exports[`GridContainer snapshot renders 1`] = `
<div
  className="MuiGrid-root makeStyles-grid-17  MuiGrid-container"
/>
`;

exports[`GridItem snapshot renders 1`] = `
<div
  className="MuiGrid-root makeStyles-grid-121  MuiGrid-item"
/>
`;

exports[`Header snapshot renders 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation4 MuiAppBar-root MuiAppBar-positionFixed makeStyles-appBar-122 undefined MuiAppBar-colorPrimary mui-fixed"
>
  <div
    className="MuiToolbar-root MuiToolbar-regular makeStyles-container-123 MuiToolbar-gutters"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text makeStyles-title-128"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      />
      <span
        className="MuiTouchRipple-root"
      />
    </button>
    <div
      className="makeStyles-flex-126"
    />
    <div
      className="PrivateHiddenCss-smDown-207"
    />
  </div>
</header>
`;

exports[`Parallax snapshot renders 1`] = `
<div
  className="makeStyles-parallax-227"
  style={
    Object {
      "backgroundImage": "url(\\"undefined\\")",
      "transform": "translate3d(0,0px,0)",
    }
  }
/>
`;

exports[`Stats snapshot renders 1`] = `
<div
  className="makeStyles-stats-228"
>
  <div
    className="MuiGrid-root makeStyles-grid-17  MuiGrid-container"
  >
    <div
      className="MuiGrid-root makeStyles-grid-121 makeStyles-statItem-230 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-icons-229"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M17 12h-5v5h5v-5zM16 1v2H8V1H6v2H5c-1.11 0-1.99.9-1.99 2L3 19c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2h-1V1h-2zm3 18H5V8h14v11z"
        />
      </svg>
      Age: undefined
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-121 makeStyles-statItem-230 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-icons-229"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M20.57 14.86L22 13.43 20.57 12 17 15.57 8.43 7 12 3.43 10.57 2 9.14 3.43 7.71 2 5.57 4.14 4.14 2.71 2.71 4.14l1.43 1.43L2 7.71l1.43 1.43L2 10.57 3.43 12 7 8.43 15.57 17 12 20.57 13.43 22l1.43-1.43L16.29 22l2.14-2.14 1.43 1.43 1.43-1.43-1.43-1.43L22 16.29z"
        />
      </svg>
      Weight: undefined
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-121 makeStyles-statItem-230 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-icons-229"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M13 6.99h3L12 3 8 6.99h3v10.02H8L12 21l4-3.99h-3z"
        />
      </svg>
      Height: undefined
    </div>
    <div
      className="MuiGrid-root makeStyles-grid-121 makeStyles-statItem-230 MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-6 MuiGrid-grid-md-3"
    >
      <svg
        aria-hidden="true"
        className="MuiSvgIcon-root makeStyles-icons-229"
        focusable="false"
        role="presentation"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 3c-4.97 0-9 4.03-9 9s4.03 9 9 9c.83 0 1.5-.67 1.5-1.5 0-.39-.15-.74-.39-1.01-.23-.26-.38-.61-.38-.99 0-.83.67-1.5 1.5-1.5H16c2.76 0 5-2.24 5-5 0-4.42-4.03-8-9-8zm-5.5 9c-.83 0-1.5-.67-1.5-1.5S5.67 9 6.5 9 8 9.67 8 10.5 7.33 12 6.5 12zm3-4C8.67 8 8 7.33 8 6.5S8.67 5 9.5 5s1.5.67 1.5 1.5S10.33 8 9.5 8zm5 0c-.83 0-1.5-.67-1.5-1.5S13.67 5 14.5 5s1.5.67 1.5 1.5S15.33 8 14.5 8zm3 4c-.83 0-1.5-.67-1.5-1.5S16.67 9 17.5 9s1.5.67 1.5 1.5-.67 1.5-1.5 1.5z"
        />
      </svg>
      Hair Color: undefined
    </div>
  </div>
</div>
`;
